repl/src/main/resources/fake_shell.py

```diff
@@ -338,6 +340,23 @@ def magic_json(name):
         'application/json': value,
     }
 
+def magic_matplot(name):
+    try:
+        value = global_dict[name]
+
+        fig = value.gcf()
+        imgdata = StringIO.StringIO()
+        fig.savefig(imgdata, format='png')
+        imgdata.seek(0)
+        encode = base64.b64encode(imgdata.buf)
+    except:
+        exc_type, exc_value, tb = sys.exc_info()
+        return execute_reply_error(exc_type, exc_value, [])
+
+    return {
+        'image/png': encode,
+        'text/plain': "",
```

Comment:
Is this really needed? I'm not really familiar with the magic stuff, but this looks a little odd.